src/Symfony/Component/Console/Style/SymfonyStyle.php

```diff
@@ -437,4 +393,56 @@ private function reduceBuffer($messages)
             return substr($value, -4);
         }, array_merge(array($this->bufferedOutput->fetch()), (array) $messages));
     }
+
+    private function createBlock($messages, $type = null, $style = null, $prefix = ' ', $padding = false, $escape = false)
+    {
+        $indentLength = 0;
+        $lines = array();
+
+        if (null !== $type) {
+            $typePrefix = sprintf('[%s] ', $type);
+            $indentLength = strlen($typePrefix);
+            $lineIndentation = str_repeat(' ', $indentLength);
+        }
+
+        // wrap and add newlines for each element
+        foreach ($messages as $key => $message) {
+            if ($escape) {
+                $message = OutputFormatter::escape($message);
+            }
+
+            $lines = array_merge($lines, explode(PHP_EOL, wordwrap($message, $this->lineLength - Helper::strlen(strip_tags($prefix)) - $indentLength, PHP_EOL, true)));
+
+            // prefix each line with a number of spaces equivalent to the type length
+            if (null !== $type) {
+                foreach ($lines as &$line) {
+                    $line = $lineIndentation === substr($line, 0, $indentLength) ? $line : $lineIndentation.$line;
+                }
+            }
+
+            if (count($messages) > 1 && $key < count($messages) - 1) {
+                $lines[] = '';
+            }
+        }
+
+        if (null !== $type) {
+            $lines[0] = substr_replace($lines[0], $typePrefix, 0, $indentLength);
+        }
+
+        if ($padding && $this->isDecorated()) {
+            array_unshift($lines, '');
+            $lines[] = '';
+        }
+
+        foreach ($lines as &$line) {
+            $line = sprintf('%s%s', $prefix, $line);
+            $line .= str_repeat(' ', $this->lineLength - Helper::strlenWithoutDecoration($this->getFormatter(), $line));
```

Comment:
> Im not sure but i guess its only needed to simulate a background style.

These two lines add spaces to each line until they have the good length, not related to `$style`.

> maybe add a little if ($line === '') { return 0; } to strlenWithoutDecoration to avoid some unnecessary calls.

It looks really good to me, but shouldn't the change be done in a specific PR? `Helper` is not touched here